src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -563,6 +563,43 @@ public static CollectionExpressionSyntax ConvertInitializerToCollectionExpressio
         }
 
         return CollectionExpression(openBracket, SeparatedList<CollectionElementSyntax>(elements), closeBracket);
+
+        CollectionElementSyntax CreateElement(ExpressionSyntax expression)
+        {
+            if (expression is InitializerExpressionSyntax { Expressions: [var keyExpression1, var valueExpression1] } initializer)
+            {
+                // If we have `{ key, ... }` we want to move the leading trivia of the `{` to the key so that it is
+                // properly indented to the same level the `{` was.
+                var openBraceAndKeyOnSingleLine = wasOnSingleLine || text.AreOnSameLine(initializer.OpenBraceToken, keyExpression1.GetLastToken());
+                if (openBraceAndKeyOnSingleLine)
+                    keyExpression1 = keyExpression1.WithLeadingTrivia(initializer.OpenBraceToken.LeadingTrivia);
+
+                // If we have `{ ..., value }` we want to move the trailing trivia of the `}` to the value to preserve trailing comments.
+                var valueAndCloseBraceOnSingleLine = wasOnSingleLine || text.AreOnSameLine(valueExpression1.GetLastToken(), initializer.CloseBraceToken);
+                if (valueAndCloseBraceOnSingleLine)
+                    valueExpression1 = valueExpression1.WithTrailingTrivia(initializer.CloseBraceToken.TrailingTrivia);
+
+                return KeyValuePairElement(keyExpression1, ColonToken.WithTriviaFrom(initializer.Expressions.GetSeparator(0)), valueExpression1);
+            }
+            else if (expression is AssignmentExpressionSyntax
+            {
+                Left: ImplicitElementAccessSyntax { ArgumentList.Arguments: [var argument] } implicitElementAccess,
+                OperatorToken: var equalsToken,
+                Right: var valueExpression2,
+            })
+            {
```

Comment:
The indentation here trip me up a bit. I expect the pattern to be indented because it is a part of the conditional. Is this a limitation of our formatter?

Reply:
Yup. It's a limitation currently. And I fully understand how that part of the formatter works:-)